proto/schema.proto

```diff
@@ -0,0 +1,170 @@
+syntax="proto3";
+
+option go_package = "github.com/kubernetes-csi/external-snapshot-metadata/pkg/grpc";
+
+service SnapshotMetadata {
+  rpc GetMetadataAllocated(GetMetadataAllocatedRequest)
+    returns (stream GetMetadataAllocatedResponse) {}
+
+  rpc GetMetadataDelta(GetMetadataDeltaRequest)
+    returns (stream GetMetadataDeltaResponse) {}
+}
+
+enum BlockMetadataType {
+  UNKNOWN=0;
+
+  // The FIXED_LENGTH value indicates that data ranges are
+  // returned in fixed size blocks.
+  FIXED_LENGTH=1;
+
+  // The VARIABLE_LENGTH value indicates that data ranges
+  // are returned in potentially variable sized extents.
+  VARIABLE_LENGTH=2;
+}
+
+// BlockMetadata specifies a data range.
+message BlockMetadata {
+  // This is the zero based byte position in the volume or snapshot,
+  // measured from the start of the object.
+  // This field is REQUIRED.
+  uint64 byte_offset = 1;
```

Comment:
The CSI spec changed to using `int64` - there are no unsigned integers in that spec!